hermes/crates/cbork/cddl-parser/src/grammar/rfc_8610.pest

```diff
@@ -103,58 +105,33 @@ int = ${ "-"? ~ uint }
 
 /// Unsigned Integers
 uint = ${
-    ( ASCII_NONZERO_DIGIT ~ ASCII_DIGIT* )
-    | ( "0x" ~ ASCII_HEX_DIGIT+ )
-    | ( "0b" ~ ASCII_BIN_DIGIT+ )
+    (ASCII_NONZERO_DIGIT ~ ASCII_DIGIT*)
+    | ("0x" ~ ASCII_HEX_DIGIT+)
+    | ("0b" ~ ASCII_BIN_DIGIT+)
     | "0"
 }
 
 /// Literal Text
 text = ${ "\"" ~ SCHAR* ~ "\"" }
 
-/// Literal Bytes - Note CDDL Spec incorrectly defines b64''.
-bytes = ${ bytes_hex | bytes_b64 | bytes_text }
-bytes_hex = ${ "h" ~ "'" ~ HEX_PAIR* ~ "'" }
-bytes_b64 = ${ "b64" ~ "'" ~ URL_BASE64 ~ "'" }
-bytes_text = ${ "'" ~ BCHAR* ~ "'" }
+/// Literal Bytes.
+bytes = ${ bsqual? ~ "'" ~ BCHAR* ~ "'" }
+bsqual = { "h" | "b64" }
 
 // -----------------------------------------------------------------------------
 // Simple multiple character sequences
 
-/// identifier,  called the `name` in the CDDL spec.
-id = ${
-    group_socket |
-    type_socket |
-    name
-}
-
-/// Special form of a name that represents a Group Socket.
-group_socket = ${ "$$" ~ ( ( "-" | "." )* ~ NAME_END )* }
-/// Special form of a name that represents a Type Socket.
-type_socket = ${ "$" ~ ( ( "-" | "." )* ~ NAME_END )* }
-/// General form of a name.
-name = ${ NAME_START ~ ( ( "-" | "." )* ~ NAME_END )* }
-
-/// A pair of hex digits. (Must always have even numbers of hex digits.)
-HEX_PAIR = _{ S ~ ASCII_HEX_DIGIT ~ S ~ ASCII_HEX_DIGIT ~ S }
-
-/// Whitespace is allowed and is ignored.
-/// This token will keep the whitespace, so it will need to handled when converted to binary.
-URL_BASE64 = _{ S ~ ( URL_BASE64_ALPHA ~ S)* ~ URL_BASE64_PAD? }
+/// identifier, called the `name` in the CDDL spec.
+id = ${ NAME_START ~ (("-" | ".")* ~ NAME_END)* }
```

Comment:
This will be a task for semantic checks to determine the type of the identifier whether it's just a `name`, `group_socket`, or `type_socket`. This would also produce fewer parsing pairs to construct an AST. And according to their ABNF, the `id` rule is simply like that, no preprocessing id categorization.

Reply:
No, it should be part of the parser.  Not a semantic check.  Again, the semantic parser should be only things the pest parser can not do.  These are part of the ABNF grammar, so they are also able to be part of the pest grammar.